### PR TITLE
Use PEP 440 prerelease tags

### LINKED ISSUE #842: beta sdist filename with extra dot.
sdist beta filename : notebook-4.1.0.b1.tar.gz

according to pep404 there should be no dot between `0` and `b1`. (does this affect upload on PyPI ?

### PR DESCRIPTION
**Pull request by parleur**
Fusion between #860 & #862. Test the version number correctness in unit test, normalize it when exec setup.py

closes #842

### CODE REVIEW

File: notebook/_version.py [hunk 1]
```diff
@@ -1,2 +1,25 @@
...
+
+"""
+
+# Downstream maintainer, when running `python.setup.py jsversion`,
+# the version string is propagated to the JavaScript files,  do not forget to
+# patch the JavaScript files in `.postN` release done by distributions. 
+
+# Next beta/alpha/rc release: The version number for beta is X.Y.ZbN **without dots**. 
+
+version_info = (4, 1, '0b1')
```
Comment (minrk) [inline notebook/_version.py]: version_info should be `(4, 1, 0, 'b1')`

File: notebook/_version.py [hunk 2]
```diff
@@ -1,2 +1,25 @@
...
+__version__ = '.'.join(map(str, version_info[:3])) + ''.join(version_info[3:])
+import re
```
Comment (minrk) [inline notebook/_version.py]: from this line down should move to the test file

File: setupbase.py [hunk 3]
```diff
@@ -65,8 +65,27 @@ def execfile(fname, globs, locs=None):
 version_ns = {}
 execfile(pjoin(repo_root, name, '_version.py'), version_ns)
 
-version = version_ns['__version__']
```
Comment (minrk) [inline setupbase.py]: The changes in this file are no longer required, since `_version.py` is setting `__version__` to a normalized value.

**Comment by Carreau** [comment]
I took care of a few of the lasts comments.

File: notebook/_version.py [hunk 4]
```diff
@@ -1,2 +1,24 @@
...
+# Downstream maintainer, when running `python.setup.py jsversion`,
+# the version string is propagated to the JavaScript files,  do not forget to
+# patch the JavaScript files in `.postN` release done by distributions. 
+
+# Next beta/alpha/rc release: The version number for beta is X.Y.ZbN **without dots**. 
+
 version_info = (4, 1, 0, 'b1')
-__version__ = '.'.join(map(str, version_info))
+__version__ = '.'.join(map(str, version_info[:3])) + ''.join(version_info[3:])
+
```
Comment (minrk) [inline notebook/_version.py]: from this line down can be removed

**Comment by minrk** [comment]
One more note about lingering unused code in `_version.py`, but otherwise +1.

File: notebook/_version.py (hunk 4 above)
Comment (Carreau) [inline notebook/_version.py]: Oops. I took care of it too.

**Comment by Carreau** [comment]
> One more note about lingering unused code in _version.py, but otherwise +1.

Thanks for your patience !

**Comment by minrk** [comment]
Thanks!

**Pull request merged**